jenkins/scripts/integration_test.sh

```diff
@@ -50,62 +43,74 @@ fi
 
 TEST_EXECUTER_FLAVOR="${TEST_EXECUTER_FLAVOR:-4C-16GB-200GB}"
 
-echo "Creating new executer VM."
-
-# Creating new port, needed to immediately get the ip
-EXT_PORT_ID="$(openstack port create -f json \
-  --network "${CI_EXT_NET}" \
-  --fixed-ip subnet="$(get_subnet_name "${CI_EXT_NET}")" \
-  "${TEST_EXECUTER_PORT_NAME}" | jq -r '.id')"
-
-# Create new executer vm
-openstack server create -f json \
-  --image "${IMAGE_NAME}" \
-  --flavor "${TEST_EXECUTER_FLAVOR}" \
-  --port "${EXT_PORT_ID}" \
-  "${TEST_EXECUTER_VM_NAME}" | jq -r '.id'
-
-# Get the IP
-TEST_EXECUTER_IP="$(openstack port show -f json "${TEST_EXECUTER_PORT_NAME}" \
-  | jq -r '.fixed_ips[0].ip_address')"
-
-echo "Waiting for the host ${TEST_EXECUTER_VM_NAME} to come up"
-#Wait for the host to come up
-wait_for_ssh "${AIRSHIP_CI_USER}" "${AIRSHIP_CI_USER_KEY}" "${TEST_EXECUTER_IP}"
-
-cat <<-EOF > "${CI_DIR}/files/vars.sh"
-REPO_ORG="${REPO_ORG}"
-REPO_NAME="${REPO_NAME}"
-REPO_BRANCH="${REPO_BRANCH}"
-UPDATED_REPO="${UPDATED_REPO}"
-UPDATED_BRANCH="${UPDATED_BRANCH}"
-CAPI_VERSION="${CAPI_VERSION}"
-CAPM3_VERSION="${CAPM3_VERSION}"
-IMAGE_OS="${IMAGE_OS}"
-DEFAULT_HOSTS_MEMORY="${DEFAULT_HOSTS_MEMORY}"
-DISTRIBUTION="${DISTRIBUTION}"
-NUM_NODES="${NUM_NODES}"
-TESTS_FOR="${TESTS_FOR}"
+DISTRIBUTION="${DISTRIBUTION:-ubuntu}"
+if [ "${DISTRIBUTION}" == "ubuntu" ]
+then
+  IMAGE_NAME="${CI_METAL3_IMAGE}"
+  BASE_VOLUME_NAME="metal3-ubuntu"
+  sh "${CI_DIR}/integration_test_setup.sh"
+else
+  IMAGE_NAME="${CI_METAL3_CENTOS_IMAGE}"
+  echo "Creating new executer VM."
+
+  # Creating new port, needed to immediately get the ip
+  EXT_PORT_ID="$(openstack port create -f json \
+    --network "${CI_EXT_NET}" \
+    --fixed-ip subnet="$(get_subnet_name "${CI_EXT_NET}")" \
+    "${TEST_EXECUTER_PORT_NAME}" | jq -r '.id')"
+
+  # Create new executer vm
+  openstack server create -f json \
+    --image "${IMAGE_NAME}" \
+    --flavor "${TEST_EXECUTER_FLAVOR}" \
+    --port "${EXT_PORT_ID}" \
+    "${TEST_EXECUTER_VM_NAME}" | jq -r '.id'
+
+  # Get the IP
+  echo "Getting the IP address of a test executer VM."
+  TEST_EXECUTER_IP="$(openstack port show -f json "${TEST_EXECUTER_PORT_NAME}" \
+    | jq -r '.fixed_ips[0].ip_address')"
+
+
+  echo "Waiting for the host ${TEST_EXECUTER_VM_NAME} to come up"
+  #Wait for the host to come up
+  wait_for_ssh "${AIRSHIP_CI_USER}" "${AIRSHIP_CI_USER_KEY}" "${TEST_EXECUTER_IP}"
```

Comment:
you could extract this in its own file too. Or you could make functions in integration_test_setup.sh, one for volume based, the other for image based deployment.